### DISCUSSION TITLE
How to add items to TabControl wrapped by UserControl in xaml.

### DISCUSSION THREAD

**Discussion by ivanov913** [AvaloniaUI/Avalonia#15474] [Q&A]
Hi all.

I was new to Avalonia or any wpf-like frameworks.

And I was trying to create a UserControl that contains a TabControl. And customize TabControl and TabItem. 
I would like this UserControl to act just like TabControl does. To add items and content like in xaml.

```xml
<local:TabControlUserControl>
	<local:CustomTabItem Header="Tab 1">
		<TextBlock Text="Content of Tab 1"/>
	</local:CustomTabItem>
	<local:CustomTabItem Header="Tab 2">
		<TextBlock Text="Content of Tab 2"/>
	</local:CustomTabItem>
</local:TabControlUserControl>
```

Is there any way to achieve this?

**Comment by timunie** [comment]
check Avalonia.Samples -> CustomControl samples

**Reply by timunie** [reply]
> [!TIP]
> You can inherit TabControl instead of UserControl here.

**Reply by ivanov913** [reply]
Thanks for reply.

I'm putting all custom controls into a class lib. 
To make user of the lib easier. I don't want to implement my custom controls as look less control.
Since it requires to add resource dictionary to App.xaml.
